tfexec/errors.go

```diff
@@ -44,9 +46,13 @@ func parseError(err error, stderr string) error {
 		return &ErrNoInit{stderr: stderr}
 	case noConfigErrRegexp.MatchString(stderr):
 		return &ErrNoConfig{stderr: stderr}
-	default:
-		return errors.New(stderr)
+	case workspaceDoesNotExistRegexp.MatchString(stderr):
+		submatches := workspaceDoesNotExistRegexp.FindStringSubmatch(stderr)
+		if len(submatches) == 2 {
```

Comment:
This feels odd to me but I have no context.

Reply:
Yeah I think at some point we will need to overhaul this from a `switch` to something a bit more sophisticated.

Reply:
I could use named groups and make it a bit cleaner possibly, but not without a lot of extra code for this single case.